packages/core/src/sync-historical/service.ts

```diff
@@ -952,115 +939,97 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
-      const chunks = getChunks({
-        intervals: [[from, to]],
-        maxChunkSize: Math.round((to - from) / 10),
-      });
-      retryRanges.push(
-        ...chunks.map(([f, t]) => [toHex(f), toHex(t)] as const),
-      );
-    } else if (
-      // Infura block range limit error.
-      error.code === LimitExceededRpcError.code &&
-      error.details!.includes("query returned more than 10000 results")
-    ) {
-      const safe = error.details!.split("Try with this block range ")[1];
-      const safeStart = Number(safe.split(", ")[0].slice(1));
-      const safeEnd = Number(safe.split(", ")[1].slice(0, -2));
-
-      retryRanges.push([toHex(safeStart), toHex(safeEnd)]);
-      retryRanges.push([toHex(safeEnd + 1), options.toBlock]);
-    } else if (
-      // Thirdweb block range limit error.
-      error.code === InvalidParamsRpcError.code &&
-      error.details!.includes("block range less than 20000")
-    ) {
-      const midpoint = Math.floor(
-        (Number(options.toBlock) - Number(options.fromBlock)) / 2 +
-          Number(options.fromBlock),
+      const retryRanges = getLogFilterRetryRanges(
+        err as LogFilterError,
+        params.fromBlock,
+        params.toBlock,
       );
 
-      retryRanges.push([toHex(options.fromBlock), toHex(midpoint)]);
-      retryRanges.push([toHex(midpoint + 1), options.toBlock]);
-    } else if (
-      // Quicknode block range limit error (should never happen).
-      error.name === "HttpRequestError" &&
-      error.details!.includes(
-        "eth_getLogs and eth_newFilter are limited to a 10,000 blocks range",
-      )
-    ) {
-      const midpoint = Math.floor(
-        (Number(options.toBlock) - Number(options.fromBlock)) / 2 +
-          Number(options.fromBlock),
-      );
-      retryRanges.push([toHex(options.fromBlock), toHex(midpoint)]);
-      retryRanges.push([toHex(midpoint + 1), options.toBlock]);
-    } else {
-      // Throw any unrecognized errors.
-      throw error;
+      return Promise.all(
+        retryRanges.map(([from, to]) =>
+          this._eth_getLogs({
+            fromBlock: from,
+            toBlock: to,
+            topics: params.topics,
+            address: params.address,
+          }),
+        ),
+      ).then((l) => l.flat());
     }
+  };
 
-    for (const [from, to] of retryRanges) {
-      const logs_ = await this._eth_getLogs({
-        ...options,
-        fromBlock: from,
-        toBlock: to,
+  /**
+   * Helper function for "eth_getBlockByNumber" request.
+   */
+  private _eth_getBlockByNumber = (params: {
+    blockNumber: number;
+  }): Promise<HistoricalBlock> =>
+    this.network.requestQueue
+      .request({
+        method: "eth_getBlockByNumber",
+        params: [numberToHex(params.blockNumber), true],
+      })
+      .then((block) => {
+        if (!block)
+          throw new BlockNotFoundError({
+            blockNumber: BigInt(params.blockNumber),
+          });
+        return block as HistoricalBlock;
       });
```

Comment:
What happens if this throws an error? Trying to get acquainted with `.then` syntax.

Reply:
If line 1002 throws an error, it gets passed up the stack because there is no `.catch()` callback. Same behavior as async/await.